archinstall/lib/disk/filesystem.py

```diff
@@ -20,24 +20,8 @@ def __init__(self, blockdevice, mode):
 		self.mode = mode
 
 	def __enter__(self, *args, **kwargs):
-		if self.blockdevice.keep_partitions is False:
```

Comment:
This looks like we no longer initialize the disk with a partition label in any case?

Reply:
That is true, it's done in the `load_layout()` instead. To reduce the complexity and separation of duty. Because the `__enter__` was doing so many things *"for no reason"*.